### PR TITLE
feat: refactor armada server chart and regenerate docs

### PR DESCRIPTION
**Pull request by dejanzele**
Refactor Armada Server chart
Regenerate docs using helm-docs

┆Issue is synchronized with this [Jira Task](https://gr-oss.atlassian.net/browse/BATCH-10) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by suprjinx — APPROVED** [review]

File: deployment/armada/values.yaml [hunk 1]
```diff
@@ -1,34 +1,126 @@
...
+# -- If a port number is specified, REST server will be exposed on that node port
+#httpNodePort: 50051
+
+# -- If specified, custom ServiceAccount name will be attached to Armada Server Deployment resource and the default ServiceAccount will not be created
+customServiceAccount: ""
+# -- More info here https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
+# -- Additional ServiceAccount properties (e.g. automountServiceAccountToken, imagePullSecrets, etc.)
+serviceAccount: {}
 
 applicationConfig:
```
Comment (severinson) [inline deployment/armada/values.yaml:86]: Do we need any config here? Check if we do anything different from the default config.

File: deployment/armada/values.yaml [hunk 2]
```diff
@@ -1,34 +1,126 @@
+nameOverride: ""
+fullnameOverride: ""
+
 image:
+  # image repository
   repository: gresearchdev/armada-server
-  tag: LATEST
+  # image tag
+  tag: v0.3.39
```
Comment (severinson) [inline deployment/armada/values.yaml:8]: Usa a variable? All components should normally be set to the same version.

File: deployment/armada/values.yaml (hunk 1 above)
Comment (dejanzele) [reply deployment/armada/values.yaml:86]: We do the following thing, which I do not think is very safe and intuitive: https://github.com/armadaproject/armada/blob/master/deployment/armada/templates/serviceaccount.yaml#L3-L11

Comment (dejanzele) [reply deployment/armada/values.yaml:86]: Do you think it is better to omit any config in the values.yaml and have docs on what is the minimal app configuration?

File: deployment/armada/values.yaml (hunk 2 above)
Comment (dejanzele) [reply deployment/armada/values.yaml:8]: What did we agree for the tag?
Should we keep LATEST here and in the Chart YAML like it was before, with the note that LATEST isn't a valid tag for Helm charts and the Chart is not well-formed?

When we release a versioned chart, we do version substitution there

File: deployment/armada/values.yaml (hunk 1 above)
Comment (severinson) [reply deployment/armada/values.yaml:86]: I think if we already have defaults in config.yaml that make sense, we shouldn't re-specify those values here. Only add to the helm files necessary overrides.

File: deployment/armada/values.yaml (hunk 2 above)
Comment (severinson) [reply deployment/armada/values.yaml]: Could we have the tag in the helm files be a variable and then set the values in the values.yaml to the latest tag when we release the charts?

Comment (dejanzele) [reply deployment/armada/values.yaml]: Helm doesn't allow templating in the Chart.yaml sadly

**Review by severinson — DISMISSED** [review]

**Review by severinson — DISMISSED** [review]

File: deployment/armada/templates/_helpers.tpl [hunk 3]
```diff
@@ -19,6 +19,28 @@
 {{- printf "%s%s" ( include "armada.users.name" .) ".yaml" -}}
 {{- end }}
 
+{{- define "armada.resource-quantity" -}}
+    {{- $value := . -}}
+    {{- $unit := 1.0 -}}
+    {{- if typeIs "string" . -}}
+        {{- $base2 := dict "Ki" 0x1p10 "Mi" 0x1p20 "Gi" 0x1p30 "Ti" 0x1p40 "Pi" 0x1p50 "Ei" 0x1p60 -}}
```
[the remark this answers is not in the document]
Comment (dejanzele) [reply deployment/armada/templates/_helpers.tpl:26]: Definitely more easy in Go than Helm, Operator handles this more gracefully

**Review by severinson — APPROVED** [review]

**Pull request merged**